pkg/logging/controller/logging/projectlogging.go

```diff
@@ -0,0 +1,102 @@
+package logging
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	loggingconfig "github.com/rancher/rancher/pkg/logging/config"
+	"github.com/rancher/rancher/pkg/logging/generator"
+	"github.com/rancher/rancher/pkg/logging/utils"
+)
+
+const (
+	ProjectIDAnnotation = "field.cattle.io/projectId"
+)
+
+type ProjectLoggingSyncer struct {
+	management           config.ManagementContext
+	k8sClient            kubernetes.Interface
+	projectLoggingClient v3.ProjectLoggingInterface
+	clusterLoggingClient v3.ClusterLoggingInterface
+}
+
+func RegisterProjectLogging(cluster *config.ClusterContext) {
+	projectLoggingClient := cluster.Management.Management.ProjectLoggings("")
+	syncer := &ProjectLoggingSyncer{
+		management:           *cluster.Management,
+		k8sClient:            cluster.K8sClient,
+		projectLoggingClient: projectLoggingClient,
+		clusterLoggingClient: cluster.Management.Management.ClusterLoggings(""),
+	}
+
+	projectLoggingClient.AddHandler("project-logging-controller", syncer.Sync)
+}
+
+func (c *ProjectLoggingSyncer) Sync(key string, obj *v3.ProjectLogging) error {
+	err := c.createOrUpdateProjectConfigMap()
```

Comment:
We should do the check in all sync methods for obj a) being nil b) having not nil deletion timestamp